src/js/libcs/Operators.js

```diff
@@ -3592,14 +3592,98 @@ evaluator.halfplane$2 = function(args, modifs) {
 //   Geometric elements      //
 ///////////////////////////////
 
-evaluator.allpoints$0 = function(args, modifs) {
+
+// helper for all*(<geoobject>)
+eval_helper.allhelper1arg = function(args, kind, restrict) {
+    if (args.length > 1) return nada;
+    var geoEl = csgeo.csnames[args[0].name];
     var erg = [];
-    for (var i = 0; i < csgeo.points.length; i++) {
-        erg[i] = {
-            ctype: "geo",
-            value: csgeo.points[i],
-            type: "P"
-        };
+    var cgeo;
+    if (geoEl) {
+        var inci = geoEl.incidences;
+        // perhaps we only want circles, segments etc ...
+        if (restrict) {
+            // restrictions are handled differently for P,L,S and C
+            if (kind === "C") { // Conics/Circles
+                inci.forEach(function(el) {
+                    if (el.matrix && el.matrix.usage !== restrict) return;
+                    cgeo = csgeo.csnames[el];
+                    if (cgeo.kind === kind) {
+                        erg.push({
+                            ctype: "geo",
+                            value: cgeo,
+                            type: cgeo.kind
+                        });
+                    }
+                });
+            } else { // point, segment, ... case
+                inci.forEach(function(el) {
+                    cgeo = csgeo.csnames[el];
+                    if (csgeo.csnames[el].type !== restrict) return;
+                    if (cgeo.kind === kind) {
+                        erg.push({
+                            ctype: "geo",
+                            value: cgeo,
+                            type: cgeo.kind
+                        });
+                    }
+                });
+            }
+        } else { // no restrictions case
+            inci.forEach(function(el) {
+                cgeo = csgeo.csnames[el];
+                if (cgeo.kind === kind) {
+                    erg.push({
+                        ctype: "geo",
+                        value: cgeo,
+                        type: cgeo.kind
```

Comment:
I know [`mover` sets the `type`](https://github.com/strobelm/CindyJS/blob/7882bf076a7baba4a810b62938b83610096fcdb1/src/js/libcs/Operators.js#L2877) (unconditionally to `P`), but [Namespace access to geo elements](https://github.com/strobelm/CindyJS/blob/7882bf076a7baba4a810b62938b83610096fcdb1/src/js/libcs/Namespace.js#L142-L145) only uses `ctype` and `value`, so that's all we should use anywhere, in my opinion.